parser/src/main/java/com/linkedin/avroutil1/util/ConfigurableAvroSchemaComparator.java

```diff
@@ -36,16 +39,45 @@
  */
 public class ConfigurableAvroSchemaComparator {
 
+  /**
+   * Compares two AvroSchema objects for equality using the specified SchemaComparisonConfiguration. This method is high
+   * performant and return on the first found mismatch.
+   *
+   * @param a The first {@link AvroSchema} to compare.
+   * @param b The second {@link AvroSchema} to compare.
+   * @param config The {@link SchemaComparisonConfiguration} that specifies the configuration for the comparison.
+   * @return true if the AvroSchema objects are equal according to the specified comparison configuration, false otherwise.
+   */
   public static boolean equals(AvroSchema a, AvroSchema b, SchemaComparisonConfiguration config) {
+    return equals(a, b, config, true).isEmpty();
+  }
+
+  /**
+   * Compares two AvroSchema objects and finds the differences between them using the specified SchemaComparisonConfiguration.
+   * This method will return a list of all the differences while comparing the schemas.
+   *
+   * @param a The first {@link AvroSchema} to compare.
+   * @param b The second {@link AvroSchema} to compare.
+   * @param config The {@link SchemaComparisonConfiguration} that specifies the configuration for the comparison.
+   * @return List of {@link AvroSchemaDifference} objects representing the differences between the two AvroSchema objects
+   */
+  public static List<AvroSchemaDifference> findDifference(AvroSchema a, AvroSchema b, SchemaComparisonConfiguration config) {
+    return equals(a, b, config, false);
+  }
+
+  private static List<AvroSchemaDifference> equals(AvroSchema a, AvroSchema b, SchemaComparisonConfiguration config, boolean fastFail) {
```

Comment:
I'm thinking that this API could be used in the future where the user just wants to find the first error. Given the current structure, we could now add an optional `failFast` parameter.

We can probably implement this as follows: create a `findDifferences()` and `findDifferencesFailFast()` methods.

Reply:
This can be a follow-up PR to unblock you

Reply:
Is there any scenario where user will need only the first diff? Considered that there is such scenario, user can get all the differences and check the first one :) thoughts?

Reply:
My thought is that this could end up being used by performance-focused use cases that still want to show a descriptive user message. They can find if the schemas are equal and if not, they can show a single error message

Reply:
Not urgent though and we can add in the future if needed so feel free to skip.